### PR TITLE
Calibcalorimetry/Calomiscalibtools: Change return type of ESProducer::produce methods to unique_ptr or shared_ptr.

### PR DESCRIPTION
**Pull request by gartung**
In preparation for a move to multi IOV events in parallel, the return type of ESProducer::produce mehtods must be changed to unique_ptr or shared_ptr. These ptrs should not refer to member data. Rather they should refer to new(copied) objects.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
This change looks good to me. It even has the advantage of decreasing our memory since the memory used by `CaloMiscalibMapEcal` is only used temporarily during the function call and not during the entire lifetime of the module.

**Comment by ghellwig** [comment]
please test

File: CalibCalorimetry/CaloMiscalibTools/plugins/CaloMiscalibTools.cc [hunk 1]
```diff
@@ -77,6 +76,7 @@ CaloMiscalibTools::~CaloMiscalibTools()
 CaloMiscalibTools::ReturnType
 CaloMiscalibTools::produce(const EcalIntercalibConstantsRcd& iRecord)
 {
+    CaloMiscalibMapEcal map_;
```
Comment (ghellwig) [inline CalibCalorimetry/CaloMiscalibTools/plugins/CaloMiscalibTools.cc]: this should be without trailing underscore as it is no data member anymore

File: CalibCalorimetry/CaloMiscalibTools/plugins/CaloMiscalibToolsMC.cc [hunk 2]
```diff
@@ -77,6 +76,7 @@ CaloMiscalibToolsMC::~CaloMiscalibToolsMC()
 CaloMiscalibToolsMC::ReturnType
 CaloMiscalibToolsMC::produce(const EcalIntercalibConstantsMCRcd& iRecord)
 {
+    CaloMiscalibMapEcal map_;
```
Comment (ghellwig) [inline CalibCalorimetry/CaloMiscalibTools/plugins/CaloMiscalibToolsMC.cc]: see above

Comment (ghellwig) [reply CalibCalorimetry/CaloMiscalibTools/plugins/CaloMiscalibToolsMC.cc]: ping

**Comment by ghellwig** [comment]
please test

**Comment by ghellwig** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**